### DISCUSSION TITLE
_TZ3000_xkap8wtb paired as TS0001_power turns off by itself like TS011F

### DISCUSSION THREAD

**Discussion by Gerd2809** [Koenkk/zigbee2mqtt#14214] [Devices]
Hej,
I use two of this devices, they have turns off by itself like the TS011F.

So I start a test.
I activate at the TS0001_power the ota update in tuya.js by insert **ota.zigbeeOTA**,
then I update the devices and after update they will resport the modellID TS011F.
After that I edit the tuya.js a second time and update the fingerprint of the TS011F din smart relay.

{
fingerprint: [{modelID: 'TS011F', manufacturerName: '_TZ3000_8bxrzyxz'},
{modelID: 'TS011F', manufacturerName: '_TZ3000_ky0fq4ho'},
**{modelID: 'TS011F', manufacturerName: '_TZ3000_xkap8wtb'}],**
model: 'TS011F_din_smart_relay',
description: 'Din smart relay (with power monitoring)',
vendor: 'TuYa', .........

Now the device work fine.

Best regards
Gerd

**Comment by KaherdinTristan** [comment]
Q: How did you update the firmware for this device? 

**Background**
- I can't get the device to display sensors for Voltage, Current or Power inside Home Assistant.

I have a Conbee 2, and to it is a "TZ3000_xkap8wtb" connected, by adding it as a "light" in "Phoscon-GW"
![image](https://user-images.githubusercontent.com/3398594/194713332-0f0dad9e-b42c-41f3-83c7-c0ea48960765.png)
No more sensors other than this on is visable in Phoscon-GW. 

In deCONZ, when the device is selected in the chart, under "Cluster info", I can see that the Voltage, Current and Power; 

![image](https://user-images.githubusercontent.com/3398594/194713266-57fe1e41-fefa-4f11-8c8c-4af6a53c9017.png)
![image](https://user-images.githubusercontent.com/3398594/194713482-99b2a862-6e27-41a7-8f9a-61d9e81ea3bb.png)

**Comment by Gerd2809** [comment]
Hej Tristan,

I use zigbee2MQTT with the zigbee-herdmans-converters, I changed the tuya.js file at the converters.
I added the line **ota:ota.zigbeeOTA** to the fingerprint of the TS1000_power in tuya.js, 
then I started the OTA update ind zigbee2MQTT.

    {
        fingerprint: [{modelID: 'TS0001', manufacturerName: '_TZ3000_xkap8wtb'}],
        model: 'TS0001_power',
        description: 'Switch with power monitoring',
        vendor: 'TuYa',
        ota: ota.zigbeeOTA,
        fromZigbee: [fz.on_off, fz.electrical_measurement, fz.metering, fz.ignore_basic_report],
        toZigbee: [tz.on_off],
        configure: async (device, coordinatorEndpoint, logger) => {
            const endpoint = device.getEndpoint(1);
            await endpoint.read('genBasic', ['manufacturerName', 'zclVersion', 'appVersion', 'modelId', 'powerSource', 0xfffe]);
            await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff', 'haElectricalMeasurement', 'seMetering']);
            await reporting.rmsVoltage(endpoint, {change: 5});
            await reporting.rmsCurrent(endpoint, {change: 50});
            await reporting.activePower(endpoint, {change: 10});
            await reporting.currentSummDelivered(endpoint);
            endpoint.saveClusterAttributeKeyValue('haElectricalMeasurement', {acCurrentDivisor: 1000, acCurrentMultiplier: 1});
            endpoint.saveClusterAttributeKeyValue('seMetering', {divisor: 100, multiplier: 1});
            device.save();
        },
        exposes: [e.switch(), e.power(), e.current(), e.voltage().withAccess(ea.STATE)],
    },
After update I change the fingerprint of TS011F
{
fingerprint: [{modelID: 'TS011F', manufacturerName: '_TZ3000_8bxrzyxz'},
{modelID: 'TS011F', manufacturerName: '_TZ3000_ky0fq4ho'},
**{modelID: 'TS011F', manufacturerName: '_TZ3000_xkap8wtb'}],**
model: 'TS011F_din_smart_relay',
description: 'Din smart relay (with power monitoring)',
vendor: 'TuYa', .........

Best regards 
Gerd

**Reply by KaherdinTristan** [reply]
Did you have to download any OTA update files first your self?
How did you edit the fingerprint?
Do you think HA ZHA also have a tuya.js file?

**Reply by GSzabados** [reply]
By adding the `ota: ota.zigbeeOTA` in the converter, it just declares that it can do a zigbeeOTA update and there might be an OTA file in the provided links.

But as you ask, has @Gerd2809 downloaded any OTA files outside of Z2M and applied it through Z2M, or just used the OTA menu and clciked on the Check for Updates and then updated it? My guess would be, the second.

But why does the device report itself after the the firmware update as TS011F, that is a good question.

The second change, I don't understand fully, why the `model: 'TS011F_din_smart_relay',` was chosen, which hasn't got the `switch_type` selection (being a DIN smart relay without a switch connection option), but has the `indicator_mode` to configure (the behavior of the LED).

Otherwise, I am quite curious, are you using the relay only with lights or with a wall socket? If wall socket, what load does it capable to handle? Vacuum cleaner? Fridge? Washing machine? Space heater?

**Comment by scoma24** [comment]
when i customize and save the tuya.js (vi tuya.js and :w) the line is saved, but after a reload of zigbee2mqtt the line is gone again. can you help me?

**Reply by scoma24** [reply]
even if I remove the write permissions from the tuya.js file (-w), it is overwritten.
